pkg/utils/wait_for_file_to_exist.go

```diff
@@ -0,0 +1,54 @@
+package utils
+
+import (
+	"os"
+	"time"
+
+	"github.com/cenkalti/backoff"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+type VerifyFileExistsFunc func(path string) error
+
+// WaitForFileToExist waits for retryCountLimit seconds to see if the file
+// exists in the given path. If it's not there by the end of the retry count limit, it returns
+// an error.
+func WaitForFileToExist(
+	path string,
+	retryCountLimit int,
+	verifyFileExistsFunc VerifyFileExistsFunc,
+) error {
+	if verifyFileExistsFunc == nil {
+		verifyFileExistsFunc = verifyFileExists
```

Comment:
Not a blocker but to make it obvious that the right-hand is a function within this module rather than a variable, you can do something like:
```go
// Top of file
var DefaultVerifyFileExistsFunc = verifyFileExists
...
func Foo() {
    if verifyFileExistsFunc == nil {
        verifyFileExistsFunc = DefaultVerifyFileExistsFunc
    }
}
//
```

Reply:
does it need to be capitalized? 

also - would it be decalred before or after `type VerifyFileExistsFunc func(path string) error`?

Reply:
> does it need to be capitalized?

No but it makes it clear that it's a constant (even if in Golang it means that it's exported) so it's up to you.

> would it be decalred before or after type VerifyFileExistsFunc func(path string) error?

probably before

Reply:
done.